test/e2e/framework/metrics/metrics_grabber.go

```diff
@@ -235,16 +237,23 @@ func (g *Grabber) Grab() (Collection, error) {
 	return result, nil
 }
 
-func (g *Grabber) getMetricsFromPod(client clientset.Interface, podName string, namespace string, port int) (string, error) {
-	rawOutput, err := client.CoreV1().RESTClient().Get().
-		Namespace(namespace).
-		Resource("pods").
-		SubResource("proxy").
-		Name(fmt.Sprintf("%v:%v", podName, port)).
-		Suffix("metrics").
-		Do(context.TODO()).Raw()
-	if err != nil {
-		return "", err
+func (g *Grabber) getMetricsFromPod(client clientset.Interface, podName string, namespace string, port int) (metrics string, err error) {
+	pollErr := wait.PollImmediate(5*time.Second, 2*time.Minute, func() (bool, error) {
+		rawOutput, err := client.CoreV1().RESTClient().Get().
+			Namespace(namespace).
+			Resource("pods").
+			SubResource("proxy").
+			Name(fmt.Sprintf("%v:%v", podName, port)).
+			Suffix("metrics").
+			Do(context.TODO()).Raw()
+		if err != nil {
+			return false, nil
+		}
+		metrics = string(rawOutput)
+		return true, nil
+	})
+	if pollErr != nil {
+		return "", pollErr
```

Comment:
this will always return "timeout waiting for condition..."

ideally the underlying error from the RESTClient call should be surfaced:
```
var err error
pollErr := wait.PollImmediate(5*time.Second, 2*time.Minute, func() (bool, error) {
.... err is set here
}
if pollErr != nil {
		return "", err
}
```

Reply:
will follow Jordan's suggestion, what do you think? his arguments sound very solid https://github.com/kubernetes/kubernetes/pull/88982/files#r389928479

Reply:
if this long poll can be avoided it's better.